Individual 3

apsitos opened the pull request:
## Purpose

_Describe the problem or feature in addition to a link to the issues. Delete any content that doesn't apply._

## Approach

_How does this change address the problem?_

### Learning

_Describe the research stage._

_Links to blog posts, patterns, libraries or add-ons used to solve this problem._

#### Blog Posts

### Open Questions and Pre-Merge TODOs

- [ ] Use Github checklists. When solved, check the box and explain the answer.

### Test coverage 

_Briefly describe what tests you have written and why._

### Merge Dependencies and Related Work

_Link to merge dependency issues or PRs._

### Follow-up tasks

- [Issue #2 (Example)](https://github.com/flexyford/pull-request/issues)

### Screenshots

#### Before

#### After

apsitos committed:
Create reminder route, add nested reminder path

apsitos committed:
Nest reminder route in reminders

apsitos committed:
Link reminder-item so when clicked the route dynamically changes

apsitos committed:
Begin testing

apsitos committed:
Add class to reminder title field to match test

apsitos committed:
Correct first assertion to match the actual route, find title of the first item within second assertion

apsitos commented:
@martensonbj The tests are not actually clicking the first item and therefore is failing the first assertion. Similarly, it is expecting an empty string for the second. When attempting to adjust the test to match the results, the found text is trimmed differently.

martensonbj commented on tests/unit/routes/reminder-test.js:
```diff
@@ -0,0 +1,11 @@
+import { moduleFor, test } from 'ember-qunit';
```
I'll let it slide but I'd like to point out that it's a bullshit test ;)

apsitos replied:
This one was automatically provided, I'm assuming by Ember during route generation. We haven't touched the unit tests, yet.

martensonbj replied:
Yep - sorry - I meant that if Ember auto generates a test you should either add legit assertions or move the test file to a different branch until you do.

martensonbj reviewed (approved):
See note about test.

martensonbj commented:
Can you clarify the problem? What I understand is that the test is not firing on the line click(".spec-reminder-item:first") and therefore failing?

apsitos commented:
Yes, sorry. Line 26 of tests/acceptance/reminder-list-test.js is being skipped over somehow. When each item is clicked on, the URL changes to /reminders/:reminders_id in the browser. But the test shows a result of "/reminders". I don't know if things are wired up wrong, or if there's something wrong in the test, but we've compared to a couple other people and they have the same thing in their test, but it passes. I'm not sure what we're missing!

martensonbj commented on app/templates/components/reminder-item.hbs:
```diff
@@ -1,6 +1,8 @@
-  <p>{{reminder.date}}</p>
-  <p>{{reminder.notes}}</p>
-  <p>{{reminder.pinned}}</p>
+  {{#link-to "reminders.reminder" reminder }}
```
Try changing this to `reminder.id`...seems like a silly workaround but in [this section of the docs](https://guides.emberjs.com/v2.11.0/routing/specifying-a-routes-model/#toc_dynamic-models) they mention the model hook not firing the same way if you pass an entire model vs the id.

martensonbj replied:
Wait I'm sorry I misread that message - the app is fine, the test isn't returning the same URL that the browser is.

martensonbj commented:
Double check the classname(s) attached to the anchor tag in your HTML - it looks like you're specifying a class name in the `reminder.js` component file, is that rendering in your browser on the anchor tag itself and not, say, a `div` wrapped around it?

apsitos committed:
Remove class from reminder-item.js and add to reminder-item.hbs file
Moving the location of the class specified in the test made the first assertion test pass.

apsitos committed:
Adjust second assertion test
Second assertion test now finds title of first item and matches it.

apsitos committed:
Comment out bullshit tests for potental future use

apsitos committed:
Add anchor tag to hold the item class, adjust tests to reflect change

apsitos committed:
Add toggleDisplay action and active property

apsitos committed:
Active class toggles

apsitos committed:
Remove unnecessary route

apsitos committed:
Add main background color

apsitos commented:
@martensonbj We've spent an hour+ trying to get the active reminder-item to render in the {{outlet}} of the reminders.hbs file (line 3). We've tried specifying {{outlet "reminder"}} (and "reminder-item"), rendering the template, etc, all to no avail. Do we need to have a controller? What are we missing?
